pkg/network/netlink/consumer.go

```diff
@@ -244,7 +244,12 @@ func (c *Consumer) DumpTable(family uint8) (<-chan Event, error) {
 		}
 	}
 
-	conn, err := netlink.Dial(unix.AF_UNSPEC, &netlink.Config{NetNS: int(c.rootNetNs)})
+	var conn *netlink.Conn
+	err = util.WithNS(c.procRoot, c.rootNetNs, func() error {
+		conn, err = netlink.Dial(unix.AF_UNSPEC, &netlink.Config{})
+		return err
+	})
```

Comment:
What's the difference between WithNS and passing the rootNS into netlink.Config?

Reply:
Passing in the namespace would cause `setns` to be called which is what we're trying to avoid. The `WithNS` function only calls `setns` if the namespace is being changed, which will happen with we are using the root ns. In this case we won't be, so `setns` is not called.